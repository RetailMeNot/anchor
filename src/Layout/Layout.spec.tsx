// REACT
import * as React from 'react';
import * as renderer from 'react-test-renderer';
// VENDOR
import { shallow } from 'enzyme';
// COMPONENT
import {
    Layout,
    DEFAULT_LAYOUT_WIDTH,
    DEFAULT_CONTENT_WIDTH,
    TRANSPARENT,
} from './Layout.component';

describe('Component: Layout', () => {
    const subject = <Layout />;
    const component = shallow(subject);

    it('should match its snapshot.', () => {
        expect(subject).toBeDefined();
        expect(component).toBeDefined();

        const tree = renderer.create(subject).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with default settings.', () => {
        expect(component.find('.anchor-layout').exists()).toBeTruthy();
        expect(component.find('StyledLayout').prop('layoutWidth')).toEqual(
            DEFAULT_LAYOUT_WIDTH
        );
        expect(
            component.find('StyledLayout').prop('layoutBackgroundColor')
        ).toEqual(TRANSPARENT);
        expect(
            component.find('StyledContent').prop('contentBackgroundColor')
        ).toEqual(TRANSPARENT);
        expect(component.find('StyledContent').prop('contentWidth')).toEqual(
            DEFAULT_CONTENT_WIDTH
        );
    });

    it('should render with passed props.', () => {
        const layoutWidth = '80px';
        const layoutBackgroundColor = '#ffffff';
        const contentWidth = '2rem';
        const contentBackgroundColor = 'orange';
        const testSubject = shallow(
            <Layout
                layoutWidth={layoutWidth}
                layoutBackgroundColor={layoutBackgroundColor}
                contentWidth={contentWidth}
                contentBackgroundColor={contentBackgroundColor}
            />
        );

        expect(testSubject.find('StyledLayout').prop('layoutWidth')).toEqual(
            layoutWidth
        );
        expect(
            testSubject.find('StyledLayout').prop('layoutBackgroundColor')
        ).toEqual(layoutBackgroundColor);
        expect(testSubject.find('StyledContent').prop('contentWidth')).toEqual(
            contentWidth
        );
        expect(
            testSubject.find('StyledContent').prop('contentBackgroundColor')
        ).toEqual(contentBackgroundColor);
    });
});
