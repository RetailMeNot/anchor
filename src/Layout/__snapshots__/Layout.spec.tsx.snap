// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Layout should match its snapshot. 1`] = `
<section
  className="anchor-layout sc-bdVaJa HHpEE"
>
  <div
    className="sc-bwzfXH ehanXF"
  />
</section>
`;
